vowpalwabbit/gd.cc

```diff
@@ -1066,7 +1069,11 @@ void save_load(gd& g, io_buf& model_file, bool read, bool text)
       save_load_online_state(all, model_file, read, text, g.total_weight, &g);
     }
     else
+    {
+      if (!all.weights.sparse && !all.weights.dense_weights.not_null())
```

Comment:
We could call `!all.weights.not_null()` and it will cover both checking for sparse and dense.